cygnus-ngsi-ld/src/main/java/com/telefonica/iot/cygnus/utils/NGSIUtils.java

```diff
@@ -0,0 +1,169 @@
+     * 
+     * @param in
+     * @param deleteSlash
+     * @param encodeSlash
+     * @return The encoded version of the input string.
+     */
+    public static String encode(String in, boolean deleteSlash, boolean encodeSlash) {
+        if (deleteSlash) {
+            return ENCODEPATTERN.matcher(in.substring(1)).replaceAll("_");
+        } else if (encodeSlash) {
+            return ENCODEPATTERN.matcher(in).replaceAll("_");
+        } else {
+            return ENCODEPATTERNSLASH.matcher(in).replaceAll("_");
+        } // if else
+    } // encode
+
+    /**
+     * Encodes a string replacing all the non alphanumeric characters by '_' (except by '-' and '.').
+     * This should be only called when building a persistence element name, such as table names, file paths, etc.
+     *
+     * @param in
+     * @return The encoded version of the input string.
+     */
+    public static String encodePostgreSQL(String in) {
+        return ENCODEPOSTGRESQL.matcher(in).replaceAll("_");
+    } // encode
+
+    /**
+     * Encodes a string replacing all '/', '\', '.', ' ', '"' and '$' by '_'.
+     * @param in
+     * @return The encoded version of the input string
+     */
+    public static String encodeSTHDB(String in) {
+        return ENCODESTHDBPATTERN.matcher(in).replaceAll("_");
+    } // encodeSTHDB
+    
+    /**
+     * Encodes a string replacing all '$' by '_'.
+     * @param in
+     * @return The encoded version of the input string
+     */
+    public static String encodeSTHCollection(String in) {
+        return ENCODESTHCOLLECTIONPATTERN.matcher(in).replaceAll("_");
+    } // encodeSTHCollection
+    
+    /**
+     * Gets a geometry value, ready for insertion in CartoDB, given a NGSI attribute value and its metadata.
+     * If the attribute is not geo-related, it is returned as it is.
+     * @param attrValue
+     * @param attrType
+     * @param metadata
+     * @param swapCoordinates
+     * @return The geometry value, ready for insertion in CartoDB/PostGIS, or the value as it is
+     */
+    public static ImmutablePair<String, Boolean> getGeometry(String attrValue, String attrType, String metadata,
+            boolean swapCoordinates) {
+        // First, check the attribute type
+        if (attrType.equals("geo:point")) {
+            String[] split = attrValue.split(",");
+                
+            if (swapCoordinates) {
+                return new ImmutablePair(
+                        "ST_SetSRID(ST_MakePoint(" + split[1].trim() + "," + split[0].trim() + "), 4326)", true);
+            } else {
+                return new ImmutablePair(
+                        "ST_SetSRID(ST_MakePoint(" + split[0].trim() + "," + split[1].trim() + "), 4326)", true);
+            } // if else
+        } // if
+        
+        if (attrType.equals("geo:json")) {
+            return new ImmutablePair("ST_GeomFromGeoJSON(" + attrValue + ")", true);
+        } // if
+
+        // TBD: What about:  ?
```

Comment:
```suggestion
        // FIXME: What about:  ?
```

(Our "convention" is to use FIXME as mark for pending stuff)